### PR TITLE
Add new plugin to display SMU firmware version on AMD APU/CPU

### PR DESCRIPTION
**Pull request by superm1**
This plugin is read-only, but can be used to tell what version of SMU firmware is installed.

This working is dependent upon:
https://patchwork.kernel.org/project/platform-driver-x86/patch/20220914141850.259-1-mario.limonciello@amd.com/

@jwrdegoede FYI so you can see the other half of this and how it's used.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/amd-pmc/fu-amd-pmc-device.c
Comment (hughsie) [inline plugins/amd-pmc/fu-amd-pmc-device.c:36]: needs braces

Comment (hughsie) [reply plugins/amd-pmc/fu-amd-pmc-device.c:36]: i also think it'd be clearer to just return FALSE early and unindent the g_propagate_error

Comment (hughsie) [reply plugins/amd-pmc/fu-amd-pmc-device.c:36]: Maybe

    if (g_error_matches(error_local, G_IO_ERROR, G_IO_ERROR_NOT_FOUND)) {
        g_set_error_literal()
        return FALSE;
    }
    g_propagate_error()
    return FALSE;

seems more glibby.

Comment (superm1) [reply plugins/amd-pmc/fu-amd-pmc-device.c]: Got it, thanks!

**Review by hughsie — APPROVED** [review]
LGTM, but maybe get the kernel ack from Hans before merging.

**Comment by superm1** [comment]
> LGTM, but maybe get the kernel ack from Hans before merging.

Yeah.  Will leave it here until it's in his review tree.

**Comment by superm1** [comment]
The kernel stuff is in hans review tree now. I'll merge this.

**Pull request merged**
